geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/kafka/commands/KeywordCommand.scala

```diff
@@ -0,0 +1,105 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.tools.kafka.commands
+
+import com.beust.jcommander._
+import com.beust.jcommander.converters.IParameterSplitter
+import org.locationtech.geomesa.kafka.{KafkaDataStore, KafkaDataStoreSchemaManager}
+import org.locationtech.geomesa.tools.kafka.commands.KeywordCommand.KeywordParameters
+import org.locationtech.geomesa.tools.common.OptionalFeatureTypeNameParam
+import org.locationtech.geomesa.tools.kafka.ProducerKDSConnectionParams
+import scala.collection.JavaConversions._
+import scala.io.StdIn
+import scala.util.{Try, Failure}
+
+class KeywordCommand(parent: JCommander) extends CommandWithKDS(parent) {
+
+  import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType._
+  override val command: String = "keyword"
+  override val params = new KeywordParameters()
+
+  override def execute(): Unit = {
+
+    if (ds == null) {
+      throw new ParameterException("Could not load a data store with the provided parameters")
+    }
+
+    val sft = ds.asInstanceOf[KafkaDataStore].getKafkaSchema(params.featureName)
+    var keywordsModified = false
+
+    if (params.keywordsToAdd != null) {
+      sft.addKeywords(params.keywordsToAdd.mkString(KEYWORDS_DELIMITER))
+      keywordsModified = true
+    }
+
+    if (params.keywordsToRemove != null) {
+      sft.removeKeywords(params.keywordsToRemove.mkString(KEYWORDS_DELIMITER))
+      keywordsModified = true
+    }
+
+    if (params.removeAll) {
+      val confirm = StdIn.readLine("Remove all keywords? (y/n): ").toLowerCase
+      if (confirm.equals("y") || confirm.equals("yes")) {
+        sft.removeAllKeywords()
+        keywordsModified = true
+      } else {
+        println("Aborting operation")
+        ds.dispose()
+        return
+      }
+    }
+
+    // Attempt to replace the old schema with the updated one
+    if (keywordsModified) {
+      Try {
+        ds.removeSchema(sft.getTypeName)
+      } match {
+        case Failure(ex) => println(ex.toString)
```

Comment:
Calling removeSchema on a KDS is risky business, as its behavior can be different depending on how Kafka is configured. In this case, what you're seeing is that the Kafka topic deletion setting for kafka isn't enabled (which is the default). If that setting is disabled, all that will happen when calling removeSchema is the Kafka topic will be "marked" for deletion (i.e. there'll be a new zk node at /admin/delete_topics/<topic_name> created indicating to Kafka that the topic is marked for deletion). On Kafka restart, the topic will be deleted, but not during actual runtime. This is something that hasn't really been addressed with the KDS... 

Either way, I don't think removeSchema is what you want to use. removeSchema on a KDS will delete the schema AND all the data associated with it. In your corresponding keywords command for Accumulo, you only call updateSchema, which doesn't remove any data. It'd probably be best for you to implement an updateSchema method for the KDS similiar to the one for the Accumulo data store (if you do, should go in KafkaDataStoreSchemaManager).